### ISSUE TITLE
Do we require a database?

### ISSUE THREAD

**Issue by jasoneflood** [jasoneflood/jaymarine#3]
Do we require a database for some services? It would be useful to be able to create a route or track times/co-ordinates/ speeds and be able to serve up a report

**Comment by markdeegan** [comment]
Initially, yes, either as a logfile on the server containing all the raw data received from agents.
Also, any responses, data, derived stats sent by the server to web pages or other subscribers etc.

**Comment by markdeegan** [comment]
We could add a logfile on the ESP, but probably best to keep that on a microSD card so the logs can be queried in the event the device fails. 
Fortunately, the cheap Adafruit displays I have here, they have a microSD slot on them.

**Comment by jasoneflood** [comment]
Ok - so you are proposing to log on the worker/agent? I can imagine these devices being installed in both hard to reach areas and water tight boxes. I think logging should be done at the coordinator level - the only thing the workers/agents should detail to the user is the LED code (connected/sending/receiving/failed) - otherwise its lots of memory cards and other hassle? If we add a log file to the worker/agent we could use it for maybe packets received (just to see what is being lost  - maybe we have a range issue or network issue - so it'll be useful for an element of debugging. But I'm not sure we want to be pulling physical cards as a norm. What do you think?

**Comment by jasoneflood** [comment]
'riffing now - but we could add a "log screen" to the worker/agent - so you don't even need to pull the card? We could put it out over HTTP or as a downloadable. (Simply trying to reduce the screwdriver and damp hands on electronics as much as possible)

**Comment by markdeegan** [comment]
Ok, avoiding screwdrivers and stuff.
Can we look at using BlueTooth to connect to the console on the devices, at least to see diagnostics.
If the device fails to connect to a WiFi access point, even though they will all fallback to the jaymarine:jaymarine netowkr, then we mihght want to get at it anyway.
Mond you, give the price-point, I think replacement with a known good device might be easier.

**Comment by jasoneflood** [comment]
Yeah - I think at the price point, its probably a case of bench repair and on site swap.  That said - we could do a bluetooth thing, but we would need to write a mobile app or something to grab the data and interpret it, or maybe there is something we can grab off the shelf to do the job.  LED flashes/beep codes don't lend themselves to easy debugging... but its a start.

I'm open to whatever you think might be practical to implement and actually useful/used in the field. Maybe we find that out during testing in the field too.
